Compile bug: arm64-windows-snapdragon-release build fails; arm64-windows-snapdragon succeeds

### Git commit

1e8924fd65ad349d1d838412a2172292618f3bbf

### Operating systems

Windows

### GGML backends

OpenCL

### Problem description & steps to reproduce


When building with the Release CMake preset `arm64-windows-snapdragon-release`, the build fails at link time with undefined symbol errors (mainCRTStartup, operator new/delete, __std_exception_*, _CxxThrowException, __CxxFrameHandler3, __acrt_iob_func, __stdio_common_vfprintf, strlen, fwrite, fputc). The linker is invoked with `-nostartfiles -nostdlib`, so the C/C++ runtime and startup are not linked.

Building with `arm64-windows-snapdragon` (no release) succeeds.

Steps (see docs/backend/snapdragon/windows.md):
1. Set env: OPENCL_SDK_ROOT, HEXAGON_SDK_ROOT, HEXAGON_TOOLS_ROOT, HEXAGON_HTP_CERT, WINDOWS_SDK_BIN.
2. Configure: `cmake --preset arm64-windows-snapdragon-release -B build-wos`
3. Build: `cmake --build build-wos`
Configure succeeds; build fails when linking executables (e.g. llama-llava-cli.exe).

## Note
Using `cmake --preset arm64-windows-snapdragon -B build-wos` (without `-release`) succeeds. Presets from CMakeUserPresets.json; doc: docs/backend/snapdragon/windows.md.


### First Bad Commit

_No response_

### Compile command

```shell
$env:OPENCL_SDK_ROOT="C:\Qualcomm\OpenCL_SDK\2.3.2"
$env:HEXAGON_SDK_ROOT="C:\Qualcomm\Hexagon_SDK\6.4.0.2"
$env:HEXAGON_TOOLS_ROOT="C:\Qualcomm\Hexagon_SDK\6.4.0.2\tools\HEXAGON_Tools\19.0.04"
$env:HEXAGON_HTP_CERT="c:\Users\MyUser\Certs\ggml-htp-v1.pfx"
$env:WINDOWS_SDK_BIN="C:\Program Files (x86)\Windows Kits\10\bin\10.0.26100.0\arm64"
cmake --preset arm64-windows-snapdragon-release -B build-wos
cmake --build build-wos
```

### Relevant log output

```shell
FAILED: bin/llama-llava-cli.exe
lld-link: error: <root>: undefined symbol: mainCRTStartup
lld-link: error: undefined symbol: void __cdecl operator delete(void *, unsigned __int64)
>>> referenced by libcpmt.lib(xthrow.obj)
lld-link: error: undefined symbol: __std_exception_copy
lld-link: error: undefined symbol: __std_exception_destroy
lld-link: error: undefined symbol: _CxxThrowException
lld-link: error: undefined symbol: const type_info::`vftable'
lld-link: error: undefined symbol: __CxxFrameHandler3
>>> referenced by tools/mtmd/deprecation-warning.cpp
lld-link: error: undefined symbol: __acrt_iob_func
lld-link: error: undefined symbol: __stdio_common_vfprintf
lld-link: error: undefined symbol: void * __cdecl operator new(unsigned __int64)
lld-link: error: undefined symbol: strlen
lld-link: error: undefined symbol: fwrite
lld-link: error: undefined symbol: fputc
clang++: error: linker command failed with exit code 1 (use -v to see invocation)

Link invocation uses -nostartfiles -nostdlib. Same for llama-minicpmv-cli.exe, llama-qwen2vl-cli.exe. ninja: build stopped: subcommand failed.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.14) # for add_link_options and implicit target directories.
+cmake_minimum_required(VERSION 3.14...3.28) # for add_link_options and implicit target directories.
 project("llama.cpp" C CXX)
 include(CheckIncludeFileCXX)
 

